apollo-integration/src/test/java/com/apollographql/apollo/ResponseWriteTestCase.java

```diff
@@ -249,35 +250,63 @@ public void operation_with_fragments() throws Exception {
     assertThat(hero.fragments().heroWithFriendsFragment().friends().get(2).fragments().humanWithIdFragment().id()).isEqualTo("1003");
     assertThat(hero.fragments().heroWithFriendsFragment().friends().get(2).fragments().humanWithIdFragment().name()).isEqualTo("Leia Organa");
 
+    final HumanWithIdFragment supermanAsHumanFragment = new HumanWithIdFragment(
```

Comment:
Yeah, I have the same thought while I was implementing write operation. I think it is user responsibility to make sure that fragments with the same id have the same data. If they are different then it depends of field tree traversal